packages/cocache/src/Cocache.js

```diff
@@ -268,7 +268,14 @@ function Cocache(maybeOptions) {
 
         return update({
           records: records.delete(id),
-          collections: collections.map((list) => list.delete(list.indexOf(id)))
+          collections: collections.map((list) => {
+            const index = list.indexOf(id);
+            if (index > 0) {
```

Comment:
This should be `index > -1`, otherwise this looks great! Don't worry though, I amended it locally.